src/main.rs

```diff
@@ -200,6 +243,26 @@ async fn async_main(config: Config) {
 					.with_context(|| format!("failed to bind udp socket {}", socket_addr))
 					.unwrap_or_else(|err| panic!("{err:?}"));
 				server.register_socket(udp_socket);
+			},
+			DownstreamConfig::Tls(downstream) => {
+				let cert_and_key =
+					load_cert_and_key(downstream.certificate, downstream.key)
+						.await
+						.with_context(|| "failed to load certificate or private key")
+						.unwrap_or_else(|err| panic!("{err:?}"));
```

Comment:
Why are you manually implementing `expect`?

Reply:
I simple like the error message more:
 with `panic`:
```
thread 'main' panicked at 'failed to load certificate or private key

Caused by:
    0: failed to open "bar"
    1: No such file or directory (os error 2)', src/main.rs:252:47
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
with `expect`:
```
thread 'main' panicked at 'failed to load certificate or private key: failed to open "bar"

Caused by:
    No such file or directory (os error 2)', src/main.rs:251:26
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

But I might should change this, to keep the code cleaner.

Reply:
Also clippy does prefer panic instead of `.expect(&format!())`

https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call